src/libraries/System.Private.CoreLib/src/System/Globalization/CalendarData.iOS.cs

```diff
@@ -0,0 +1,111 @@
+            this.saSuperShortDayNames = GetCalendarInfoNative(localeName, calendarId, CalendarDataType.SuperShortDayNames).Split("||");
+
+            string? leapHebrewMonthName = null;
+            this.saMonthNames = NormalizeMonthArray(GetCalendarInfoNative(localeName, calendarId, CalendarDataType.MonthNames).Split("||"), calendarId, ref leapHebrewMonthName);
+            if (leapHebrewMonthName != null)
+            {
+                Debug.Assert(this.saMonthNames != null);
+
+                // In Hebrew calendar, get the leap month name Adar II and override the non-leap month 7
+                Debug.Assert(calendarId == CalendarId.HEBREW && saMonthNames.Length == 13);
+                saLeapYearMonthNames = (string[]) saMonthNames.Clone();
+                saLeapYearMonthNames[6] = leapHebrewMonthName;
+
+                // The returned data has 6th month name as 'Adar I' and 7th month name as 'Adar'
+                // We need to adjust that in the list used with non-leap year to have 6th month as 'Adar' and 7th month as 'Adar II'
+                // note that when formatting non-leap year dates, 7th month shouldn't get used at all.
+                saMonthNames[5] = saMonthNames[6];
+                saMonthNames[6] = leapHebrewMonthName;
+
+            }
+            this.saAbbrevMonthNames = NormalizeMonthArray(GetCalendarInfoNative(localeName, calendarId, CalendarDataType.AbbrevMonthNames).Split("||"), calendarId, ref leapHebrewMonthName);
+            this.saMonthGenitiveNames = NormalizeMonthArray(GetCalendarInfoNative(localeName, calendarId, CalendarDataType.MonthGenitiveNames).Split("||"), calendarId, ref leapHebrewMonthName);
+            this.saAbbrevMonthGenitiveNames = NormalizeMonthArray(GetCalendarInfoNative(localeName, calendarId, CalendarDataType.AbbrevMonthGenitiveNames).Split("||"), calendarId, ref leapHebrewMonthName);
+
+            this.saEraNames = NormalizeEraNames(calendarId, GetCalendarInfoNative(localeName, calendarId, CalendarDataType.EraNames).Split("||"));
+            this.saAbbrevEraNames = Array.Empty<string>();//NormalizeEraNames(calendarId, GetCalendarInfoNative(localeName, calendarId, CalendarDataType.AbbrevEraNames).Split("||"));
+
+            return this.sNativeName != null && this.saShortDates != null && this.saLongDates != null && this.saYearMonths != null &&
+                   this.saDayNames != null && this.saAbbrevDayNames != null && this.saSuperShortDayNames != null && this.saMonthNames != null &&
+                   this.saAbbrevMonthNames != null && this.saMonthGenitiveNames != null && this.saAbbrevMonthGenitiveNames != null &&
+                   this.saEraNames != null && this.saAbbrevEraNames != null;
+        }
+
+        private static string[] NormalizeEraNames(CalendarId calendarId, string[]? eraNames)
+        {
+            // .NET expects that only the Japanese calendars have more than 1 era.
```

Comment:
Should we add a check to ensure that? Or is there some check in place somewhere that covers this.

Reply:
For checking Japanese calendar era count? That we don't have, but as ICU also doesn't have I am not sure if I need to throw exception if `<=1`.